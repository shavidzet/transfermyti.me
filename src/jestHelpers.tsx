/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
