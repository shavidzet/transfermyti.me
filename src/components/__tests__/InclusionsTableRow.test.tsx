import React from "react";
import { render, RenderResult } from "~/jestHelpers";
import InclusionsTableRow from "../InclusionsTableRow";

const setup = (
  propOverrides: any = {},
): { props: any; wrapper: RenderResult } => {
  const props = {
    disabled: false,
    ...propOverrides,
  };

  const wrapper = render(
    <table>
      <tbody>
        <InclusionsTableRow {...props} />
      </tbody>
    </table>,
  );

  return { props, wrapper };
};

describe("the <InclusionsTableRow> component", () => {
  test("matches its snapshot when props.disabled = true", () => {
    const { wrapper } = setup({ disabled: true });

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });

  test("matches its snapshot when props.disabled = false", () => {
    const { wrapper } = setup({ disabled: false });

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });
});
