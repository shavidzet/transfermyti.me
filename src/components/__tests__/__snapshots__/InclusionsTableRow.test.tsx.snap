// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <InclusionsTableRow> component matches its snapshot when props.disabled = false 1`] = `
.emotion-0 td {
  color: rgb(72,72,72);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<table>
  <tbody>
    <tr
      class="emotion-0"
    />
  </tbody>
</table>
`;

exports[`the <InclusionsTableRow> component matches its snapshot when props.disabled = true 1`] = `
.emotion-0 td {
  color: rgb(136,136,136);
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

<table>
  <tbody>
    <tr
      class="emotion-0"
    />
  </tbody>
</table>
`;
